lib/review/review.controllers.js

```diff
@@ -1,53 +1,210 @@
+
+      if (existingReview !== null) {
+        throw Boom.badRequest('This Account has already created a Review for this Item');
+      }
+
+      let reviewObject = {
+        AuthorId: account.id,
+        subject: request.payload.subject,
+        body: request.payload.body,
+        rating: request.payload.rating,
+        reviewableType: reviewableType,
+        reviewableId: reviewableId,
+      };
+
+      let review;
+      try {
+        review = await createReview(reviewObject);
+      } catch (e) {
+        throw Boom.badImplementation('Error during createReview(reviewObject). ' + e);
+      }
 
+      return review;
     },
     postPropertyReview: async function (request, h) {
+      let reviewableType = 'property';
+      let reviewableId = request.params.id;
+
+      let account;
+      try {
+        account = await accountLib.getAccount(request.auth.credentials);
+        console.log(account);
+      } catch (e) {
+        throw Boom.badImplementation('Error during getAccount(request.auth.credentials). ' + e);
+      }
+
+      if (account === null) {
+        throw Boom.badRequest('Must have an Account to create a Review');
+      }
+
+      let existingReview;
+      try {
+        existingReview = await getOneAccountItemReview(
+          account.id,
+          reviewableId,
+          reviewableType
+        );
+      } catch (e) {
+        throw Boom.badImplementation('Error during getAccountItemsReviews(...). ' + e);
+      }
+
+      if (existingReview !== null) {
+        throw Boom.badRequest('This Account has already created a Review for this Item');
+      }
+
+      let reviewObject = {
+        AuthorId: account.id,
+        subject: request.payload.subject,
+        body: request.payload.body,
+        rating: request.payload.rating,
+        reviewableType: reviewableType,
+        reviewableId: reviewableId,
+      };
+
+      let review;
+      try {
+        review = await createReview(reviewObject);
+      } catch (e) {
+        throw Boom.badImplementation('Error during createReview(reviewObject). ' + e);
+      }
 
+      return review;
     },
-    deleteReview: function (request, h){
-      return models.reviews.destroy({
-        where: {
-          id: request.params.id,
-        },
-      })
-      .then(response => {
-        return h
-        .response(response)
-        .code(202);
-      })
-      .catch(error => {
-        return error;
-      });
+    deleteReview: function (request, h) {
+      // return models.reviews.destroy({
+      //   where: {
+      //     id: request.params.id,
+      //   },
+      // });
+      // .then(response => {
+      //   return h
+      //   .response(response)
+      //   .code(202);
+      // })
+      // .catch(error => {
+      //   return error;
+      // });
     },
   };
 
-  function getReviews(reviewedItemId) {
-    return models.Review.findAll({
-      where: {
-        id: reviewedItemId,
-      }
+  function getReviewOptions() {
```

Comment:
when finding and returning Reviews from the database we never want to leak the AuthorId, to protect anonymity.